### PR TITLE
XS - EPIC: Improve usability of the DriverAvailabilities Create Page

### LINKED ISSUE #11: EPIC: Improve usability of the DriverAvailabilities Create Page
_### As a...

driver

### I can...

Undestand what each of the form fields mean

### So that...

i can submit a correct availability without running into issues

### Discussion

The current create page for DriverAvailabilities has a number of aspects that should be improved.

First:
When create an availability we do not need to take the driver id as input, rather it should be automatically assigned to the creator.

Second:
image
The form labels are very unprofessional and jarring. We need to change this to be consistent with the rest of the application. Please capitalize and separate words.

Third:
image
Input validation error messages on bad input also must be improved with capitalization and spaces.

Fourth:
image
There is no 'bad-input' error messages. We should have error messages on bad input for `Drive Id`, `Day`, `Start Time`, `End Time`. For day it would be ideal to convert this field to a dropdown, similar to the one on the `RideRequestForm`.

Fifth
Like the `RideRequestForm` we want to have example messages on each of the form fields to help making it easier for the driver to know how to fill it out
Example:
image

### Acceptance Criteria

- [x] The form labels are improved
- [x] The bad input and missing input error messages are added and improved
- [x] Example inputs on each form field

### Implementation Todos

Issue 1 : Backend:
- [x] Modify `DriverAvailabilityController.java` `POST` so that when creating a new `DriverAvailability` we automatically use the creator's id and no longer take in id as an input.
- [x] Modify `DriverAvailabilityController.java` `PUT` so that when editing `DriverAvailability` we make sure that the `DriverAvailability` we intend to edit is really owned by the current user, and we no longer want to set the driver id to the incoming.
- [x] Add the proper validations to `GET` single and `DELETE` so that a user can only delete or get their own availabilities and not anyone elses

Issue 2 : Frontend:
- [x] Modify `DriverAvailabilityForm.js` to meet acceptance criterea.

You may further split this into more issues if deemed necessary

### PR DESCRIPTION
**Pull request by xinyao-song**
## Dokku: https://gauchoride-qa.dokku-16.cs.ucsb.edu

(Deployed on gauchoride-qa as gauchoride-xinyao-dev is taken for PR #23 concurrently)

The PR modifies the `DriverAvailabilitiesForm` page so a driver can understand what each field in the `Create Availability Form` means and submit a correct availability without running into issues. Furthermore, the PR removed the necessity for drivers to input their `driver ID` each time they submit an availability form where the `driver ID` is automatically recorded. In this case, most number of changed lines is due to a combination of having to touch related documentation to the `driverId` field and adding new tests. 

### Testing Instructions:
1. Click `Admin` in the nav bar and go the `Users` page to toggle `Driver` under your name. 
2. Once `Driver` is being toggled, click `Shifts` in the nav bar and go to the `Availability` page. 
3. From there you can see a button called `Create Driver Availability` and that is where you can fill out a driver availability form and test out the functionality of this PR. 

### Acceptance Criteria:

- [ ] The form labels are improved so that each word are spaced out and is easily readable to the user. 
- [ ] The bad input and missing input error messages are added for `Start Time`, `End Time`, and `Day of the Week`. 
- [ ] There are example inputs on each form field which helps the user to know what to put there. 
- [ ] `DriverId` is removed from all aspects of DriverAvailability, including `DriverAvailabilityController.java`, `DriverAvailabilityForm.js`, `DriverAvailabilityEditPage.js`, and `DriverAvailabilityCreatePage.js`. 
- [ ] User is able to input additional notes in the `notes` field if there are anything additional to say. 

#### Before:
image
image
image

#### After: 
Screenshot 2024-05-25 at 8 36 45 PM
Screenshot 2024-05-25 at 8 36 52 PM
 
**Example of driver's availability created without additional notes (id:8):** 
Screenshot 2024-05-25 at 8 37 08 PM

**Example of driver's availability created with additional notes (id:9):** 
Screenshot 2024-05-25 at 8 37 39 PM
Closes #11

### CODE REVIEW

**Review by spark686 — DISMISSED** [review]
LGTM

**Review by Hannahshakouri01 — DISMISSED** [review]
nice work, LGTM!

**Review by pconrad — COMMENTED** [review]
This is a lot of work!  Accordingly I think I want to test it on a dokku deployment before I'd merge it.

**Comment by xinyao-song** [comment]
Added Dokku deployment link in the PR description along with testing instructions

File: frontend/src/main/components/Driver/DriverAvailabilityForm.js [hunk 1]
```diff
@@ -2,131 +2,141 @@ import { Button, Form } from 'react-bootstrap';
...
-        </Form>
+      <Form.Group className="mb-3">
+        <Form.Label htmlFor="notes">Notes</Form.Label>
+        <Form.Control
+          data-testid={testIdPrefix + "-notes"}
+          id="notes"
+          type="text"
+          placeholder="Include any additional notes here."
+          isInvalid={Boolean(errors.notes)}
+          {...register("notes", {
```
Comment (pconrad) [inline frontend/src/main/components/Driver/DriverAvailabilityForm.js]: Consider making this optional.  I don't think making this required is going to be a good user experience.

**Review by pconrad — CHANGES_REQUESTED** [review]
I think the PR description could be better.. I made some specific suggestions in the team slack channel.

Also: I wonder why description is required?  Seems like that should be optional.

Otherwise, the code looks really good.

**Review by pconrad — DISMISSED** [review]
LGTM

**Pull request merged**